app/src/main/java/net/gsantner/markor/format/csv/CsvTextConverter.java

```diff
@@ -0,0 +1,186 @@
+        public static String toMdTable(String csvMarkup) {
+            // parser cannot handle empty lines if they are not "\r\n"
+            return toMdTable(new StringReader(csvMarkup.replace("\n", "\r\n")));
+        }
+
+        public static String toMdTable(Reader csvMarkup) {
+            StringBuilder mdMarkup = new StringBuilder();
+            try (BufferedReader bufferedReader = new BufferedReader(csvMarkup, BUFFER_SIZE)) {
+                CsvConfig csvConfig = inferCsvConfiguration(bufferedReader);
+                try (Csv2MdTable toMdTable = new Csv2MdTable(csvConfig, bufferedReader)) {
+                    String[] headers = toMdTable.readNextCsvColumnLine();
+
+                    if (headers != null && headers.length > 0) {
+                        addColumnsLine(mdMarkup, headers, headers.length);
+
+                        for (int i = 0; i < headers.length; i++) {
+                            mdMarkup.append(MD_HEADER_LINE_DELIMITER);
+                        }
+                        mdMarkup.append(MD_COL_DELIMITER).append(MD_LINE_DELIMITER);
+
+                        String[] lineColumns;
+                        while (null != (lineColumns = toMdTable.readNextCsvColumnLine())) {
+                            addColumnsLine(mdMarkup, lineColumns, headers.length);
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                // invalid csv format in editor should not crash the app when trying to display as html
+                // openCsv-3.10 may throw IOException or RuntimeExcpetion,
+                // openCsv-5.7 may throw IOException or CsvValidationException.
+                e.printStackTrace();
+            }
+            return mdMarkup.toString();
+        }
+
+        private static CsvConfig inferCsvConfiguration(BufferedReader bufferedReader) throws IOException {
+            // remember where we started.
+            bufferedReader.mark(BUFFER_SIZE);
+            try {
+                String line;
+                while (null != (line = bufferedReader.readLine())) {
+                    line = line.trim();
+                    if (!line.isEmpty() && !line.startsWith("#")) {
+                        return CsvConfig.infer(line);
+                    }
+                }
+                return CsvConfig.DEFAULT;
+            } finally {
+                // go back to start of csv
+                bufferedReader.reset();
+            }
+        }
+
+        private static void addColumnsLine(StringBuilder mdMarkup, String[] columns, int headerLength) {
+            for (int i = 0; i < max(headerLength, columns.length); i++) {
+                addColumnContainingNL(mdMarkup.append(MD_COL_DELIMITER), getCol(columns, i));
+            }
+            mdMarkup.append(MD_COL_DELIMITER).append(MD_LINE_DELIMITER);
+        }
+
+        private static String getCol(String[] columns, int i) {
+            if (i >= 0 && i < columns.length) return columns[i];
```

Comment:
Please use i.e. `return condition ? a : b`. Inline if with a action afterwards is prone to writing bad code

Reply:
done